be/src/formats/parquet/file_writer.cpp

```diff
@@ -496,20 +496,20 @@ Status AsyncFileWriter::close(RuntimeState* state,
 
         try {
             _writer->Close();
+            _file_metadata = _writer->metadata();
         } catch (const ::parquet::ParquetStatusException& e) {
             LOG(WARNING) << "close writer error: " << e.what();
             set_io_status(Status::IOError(fmt::format("{}: {}", "close writer error", e.what())));
         }
-        _chunk_writer = nullptr;
 
-        _file_metadata = _writer->metadata();
         auto st = _outstream->Close();
         if (!st.ok()) {
             LOG(WARNING) << "close output stream error: " << st.message();
             set_io_status(Status::IOError(fmt::format("{}: {}", "close output stream error", st.message())));
         }
 
-        if (cb != nullptr) {
+        // invoke callback if status is ok
+        if (cb != nullptr && get_io_status().ok()) {
```

Comment:
My question is if someday this callback function will be called even when get_io_status() is not ok, what will happen? Are you sure if the developer who will change it knows the dependence you added here?

Reply:
- The callback function collects stats of each file, and there is no need to invoke it if the io status is not ok. 
- I put `DCHECK(writer->metadata() != nullptr);` in the callback to state my intention.

Reply:
My suggestion is why not make callback function have return value other than void

Reply:
👀 Let's merge this patch, and I will refactor this part this week.